src/index.ts

```diff
@@ -144,6 +150,17 @@ class Client {
             metric += '|@' + sampling;
         }
 
+        if (this.tags) {
+            if (Array.isArray(this.tags)) {
+                metric += `|#${this.tags.join(',')}`;
+            } else {
+                const tags = Object.keys(this.tags)
```

Comment:
What do you think if we precompute the tags string? We could move this logic in the Constructor, maintaining only the string concatenation here.